hack/bundle-gen.py

```diff
@@ -0,0 +1,429 @@
+    subprocess.run(cmd)
+    print()
+
+    # commit files
+    print("Adding file")
+    repo.git.add(HIVE_SUB_DIR)
+
+    print("Commiting {}".format(pr_title))
+    try:
+        repo.git.commit('--signoff', '--message="{}"'.format(pr_title))
+    except:
+        print("Failed to commit")
+        raise
+    print()
+
+    if not dry_run:
+        print("Pushing branch {}".format(branch_name))
+        origin = repo.git.origin
+        try:
+            origin.push('--force', branch_name)
+        except:
+            print("failed to push branch to origin")
+            raise
+
+        # open PR
+        gh = importlib.import_module('github')
+        client = gh.GitHubClient(dest_github_org, dest_github_reponame, "")
+
+        from_branch = "{}:{}".format(gh_username, branch_name)
+        to_branch = "main"
+
+        body = pr_title
+        if hold:
+            body = "%s\n\n/hold" % body
+
+        resp = client.create_pr(from_branch, to_branch, pr_title, body)
+        if resp.status_code != 201: #201 == Created
+            print(resp.text)
+            sys.exit(1)
+
+        json_content = json.loads(resp.content.decode('utf-8'))
+        print("PR opened: {}".format(json_content["html_url"]))
+
+    else:
+        print("Skipping branch push due to dry-run")
+    print()
+
+def branch_is_on_master_commit(branch, repo_dir):
+    repo = git.Repo(repo_dir)
+    return repo.rev_parse('master') == repo.rev_parse(branch)
+
+def buildah_installed():
+    return shutil.which('buildah') is not None
+
+if __name__ == "__main__":
+    args = get_params()
+
+    if not buildah_installed():
+        print('buildah not found, please install buildah')
+        sys.exit(1)
+
+    update_channels = [CHANNEL_DEFAULT]
+    if args.branch:
+        if branch_is_on_master_commit(args.branch, os.getcwd()):
+            update_channels.append(args.branch)
+        else:
+            update_channels = [args.branch]
+
+    hive_version = ""
+    bundle_dir = tempfile.TemporaryDirectory(prefix='hive-operator-bundle-')
+    work_dir = tempfile.TemporaryDirectory(prefix='operatorhub-push-')
+
+    if CHANNEL_DEFAULT in update_channels:
+        hive_version = gen_hive_version(os.getcwd(), BRANCH_DEFAULT, 'HEAD', HIVE_VERSION_PREFIX)
+        build_and_push_image(os.getcwd(), args.registry_auth_file, hive_version, BRANCH_DEFAULT, args.dry_run)
+        previous_hive_version = get_previous_version(CHANNEL_DEFAULT)
+        generate_csv_base(bundle_dir.name, hive_version, previous_hive_version, OPERATORHUB_HIVE_IMAGE_DEFAULT, CHANNEL_DEFAULT)
+    else:
+        previous_hive_version = get_previous_version(args.branch)
+        # the version prefix is determined by the previous version when the
+        # provided branch does not correspond with the master branch
+        # eg. previous_hive_version = "hive-operator.v1.1.9-1.1.8.2" -> hive_version_prefix = "v1.1"
+        hive_version_prefix = previous_hive_version.replace('hive-operator.', '')[0:4]
```

Comment:
This would break down when we hit e.g. `v1.10`. Perhaps
```suggestion
        hive_version_prefix = previous_hive_version.replace('hive-operator.', '').split('.', 2)[0..1].join('.')
```
ish?

Reply:
Oh, wait. I think we want the prefix to come from the branch/channel name, not the version tag.